hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/DumpContractBytecodesSubcommand.java

```diff
@@ -0,0 +1,230 @@
+        }
+
+        final var sb = new StringBuilder(estimateReportSize(contractsWithBytecode));
+        if (emitSummary == DumpStateCommand.EmitSummary.YES) {
+            sb.append("%d registered contracts, %d with bytecode (%d are zero-length)%s, %d deleted contracts%n"
+                    .formatted(
+                            totalContractsRegisteredWithAccounts,
+                            totalContractsPresentInFileStore + zeroLengthContracts.size(),
+                            zeroLengthContracts.size(),
+                            uniqify == DumpStateCommand.Uniqify.YES
+                                    ? ", %d unique (by bytecode)".formatted(totalUniqueContractsPresentInFileStore)
+                                    : "",
+                            contractsWithBytecode.deletedContracts().size()));
+        }
+        appendFormattedContractLines(sb, contractsWithBytecode);
+
+        if (verbosity == Verbosity.VERBOSE) System.out.printf("=== Have %d byte report%n", sb.length());
+
+        writeReportToFile(sb.toString());
+    }
+
+    int estimateReportSize(@NonNull Contracts contracts) {
+        int totalBytecodeSize = contracts.contracts().stream()
+                .map(Contract::bytecode)
+                .mapToInt(bc -> bc.length)
+                .sum();
+        // Make a swag based on how many contracts there are plus bytecode size
+        int reportSizeEstimate = contracts.registeredContractsCount() * 75 + totalBytecodeSize * 2;
+        if (verbosity == Verbosity.VERBOSE) System.out.printf("=== Estimating %d byte report%n", reportSizeEstimate);
+        return reportSizeEstimate;
+    }
+
+    void writeReportToFile(@NonNull String report) {
+        try (final PrintWriter out = new PrintWriter(bytecodePath.toFile(), StandardCharsets.UTF_8)) {
+            out.print(report);
+            out.flush();
+        } catch (final FileNotFoundException ex) {
+            System.err.printf("*** Cannot create '%s' for writing%n", bytecodePath);
+        } catch (final IOException ex) {
+            System.err.printf("*** Exception when trying to write '%s'%n", bytecodePath);
+            throw new UncheckedIOException(ex); // This is a CLI program: Java will print the stack trace properly
+        }
+    }
+
+    /** Returns all _unique_ contracts (by their bytecode) from the signed state.
+     *
+     * Returns the set of all unique contracts (by their bytecode), each contract bytecode with _all_ of the
+     * contract ids that have that bytecode. Also returns the total number of contracts registered in the signed
+     * state.  The latter number may be larger than the number of contracts-with-bytecodes because some contracts
+     * known to accounts are not present in the file store.  Deleted contracts are _omitted_.
+     */
+    @NonNull
+    Pair<Contracts, List<Integer>> uniqifyContracts(
+            @NonNull final Contracts contracts, @NonNull final List<Integer> zeroLengthContracts) {
```

Comment:
why is zeroLengthContracts passed in?  It doesn't look like it is being used within the method but only to create the return value.

Reply:
that's why.  I need it on the way out for the return value, which needs to be the same type (that `Pair`) - I mean, it doesn't _need_ to be but it makes the code easier to read at the _point of call_.  And, personal preference, I don't particularly like using class fields as "global variables" even though that's standard O-O practice.  Practically all the methods in this class could be declared `static` in fact, but I also don't like that.

Damn, there's a lot of things I don't like.  People must think I'm kind of a grouchy jerk ....  I do point out things on code reviews that I _do like_!  Maybe that makes up for it?